### PR TITLE
chore: add firebase setup for iOS

### PR DESCRIPTION
**Pull request by xiangxinyue**
Changes

add firebases set-up file and upgrade the dependency things.
UI

No visible changes.
Notes

https://flutter.dev/docs/development/ios-project-migration

Important to look at this migration documentation when upgrading the flutter for iOS.

### CODE REVIEW

File: mentorApp/ios/Runner/GoogleService-Info.plist [hunk 1]
```diff
@@ -0,0 +1,36 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (HelenGezahegn) [inline mentorApp/ios/Runner/GoogleService-Info.plist:1]: [Looking at step 3.2 in the instructions to add firebase to a flutter app](https://firebase.google.com/docs/flutter/setup?platform=ios), it says to add this file in the `Runner/Runner` directory. Was it not working when you did that?

File: mentorApp/lib/main.dart [hunk 2]
```diff
@@ -2,8 +2,8 @@ import 'package:device_simulator/device_simulator.dart';
 import 'package:flutter/material.dart';
 import "./Navigation/Root.dart";
 
-const bool debugEnableDeviceSimulator = false;
-
+//const bool debugEnableDeviceSimulator = false;
```
Comment (HelenGezahegn) [inline mentorApp/lib/main.dart:5]: Please delete this comment.

File: mentorApp/ios/Runner/GoogleService-Info.plist (hunk 1 above)
Comment (xiangxinyue) [reply mentorApp/ios/Runner/GoogleService-Info.plist:1]: Using Xcode to do it works for me.

File: mentorApp/lib/main.dart (hunk 2 above)
Comment (xiangxinyue) [reply mentorApp/lib/main.dart:5]: Ok.

File: mentorApp/ios/Runner/GoogleService-Info.plist (hunk 1 above)
Comment (HelenGezahegn) [reply mentorApp/ios/Runner/GoogleService-Info.plist:1]: Wait, so did they have the `Runner/Runner` directory??

File: mentorApp/lib/main.dart (hunk 2 above)
Comment (HelenGezahegn) [reply mentorApp/lib/main.dart:5]:
```suggestion
```

File: mentorApp/ios/Runner/GoogleService-Info.plist (hunk 1 above)
Comment (xiangxinyue) [reply mentorApp/ios/Runner/GoogleService-Info.plist:1]: YES

Comment (xiangxinyue) [reply mentorApp/ios/Runner/GoogleService-Info.plist:1]: Open the iOS part of the whole project in the Xcode.

Comment (HelenGezahegn) [reply mentorApp/ios/Runner/GoogleService-Info.plist:1]: Oh, ok! Should we move this file to that directory instead? It says in the instructions to move it to `Runner/Runner` and we currently have it in `/Runner`.

Comment (HelenGezahegn) [reply mentorApp/ios/Runner/GoogleService-Info.plist:1]: As discussed offline, when we open xCode, we see that this file is located in `Runner/Runner`. Good job Rex! 👍

Comment (xiangxinyue) [reply mentorApp/ios/Runner/GoogleService-Info.plist:1]: Thank you Helen!

**Pull request merged**